src/filehandler/ifilehandler.cpp

```diff
@@ -18,17 +19,17 @@ bool IFileHandler::fileExists() const
     return f.good() && f.is_open();
 }
 
-bool IFileHandler::isFileExtension(std::string const ext) const
+bool IFileHandler::isFileExtension(std::string const &ext) const
 {
     return getFileExtension() == ext;
 }
 
 std::string IFileHandler::getFileExtension() const
 {
-    std::string ext = "";
+    std::string ext;
 
-    if (m_filePath.find_last_of(".") != std::string::npos)
-        ext = m_filePath.substr(m_filePath.find_last_of(".") + 1);
+    if (m_filePath.find_last_of('.') != std::string::npos)
```

Comment:
Sorry for nitpicking - perhaps add a temporary variable called `dot_position` - not for optimisation, of course, but for readability :)

Reply:
No prob, good point though !
Added a temporary variable here, as you suggested !